services/eth_gateway.go

```diff
@@ -234,16 +233,16 @@ func getCurrentBlock() (*types.Block, error) {
 }
 
 // WaitForTransfer is blocking call that will observe on brokerAddr on transfer on ETH.
-// If it is completed return true, otherwise, return false (or time-out)
-func waitForTransfer(brokerAddr common.Address) (bool, error) {
+// If it is completed return number of PRL.
```

Comment:
Number fo PRLs should be a decimal since we can have partial PRLs (eg: 1.5)

Reply:
Confusing, since part of our code use big.Int, part of our code use decimal. We should just use one

Reply:
Currently we're using decimal for the Cost/Invoice, but we could certainly change that.  This might be a dumb question but do big.Ints let you have a decimal place?  Or do they just show a much smaller unit of the currency?  

If we have to send 1.5 PRL I'd rather it show up as 1.5 PRL for readability, would be my personal preference.

Reply:
I was thinking that since we know PRLs have 18 decimal precision, we would store this as an int, then divide by 10^18 we want to convert PRLs. It complicates things a bit.

If Golang has arbitrary precision decimals (or at least to 18 decimals), I would prefer to use that so we don't get confused with the conversion.

I'm going to approve this PR, and we can convert everything to something consistent in a future PR.

Reply:
Looks like we can use this: https://golang.org/pkg/math/big/#NewFloat

Reply:
https://github.com/shopspring/decimal#why-dont-you-just-use-bigrat
Probably to stick with decimal

Reply:
SGTM